diff --git a/browser/base/content/navigator-toolbox.inc.xhtml b/browser/base/content/navigator-toolbox.inc.xhtml
index 30e7b9b2ac63db6ccd2727a9341081cecefc25cb..fa2b32afce8064ac372aad78c6ff9e2e68a4f865 100644
--- a/browser/base/content/navigator-toolbox.inc.xhtml
+++ b/browser/base/content/navigator-toolbox.inc.xhtml
@@ -2,7 +2,7 @@
 # License, v. 2.0. If a copy of the MPL was not distributed with this
 # file, You can obtain one at http://mozilla.org/MPL/2.0/.
 
-<toolbox id="navigator-toolbox" class="browser-toolbox-background">
+<toolbox id="navigator-toolbox" class="browser-toolbox-background" persist="width style">
   <script src="chrome://browser/content/navigator-toolbox.js" />
 
   <!-- Menu -->
@@ -17,9 +17,9 @@
 #include browser-menubar.inc
     </toolbaritem>
     <spacer flex="1" skipintoolbarset="true" style="order: 1000;"/>
-#include titlebar-items.inc.xhtml
   </toolbar>
 
+<hbox id="titlebar">
   <toolbar id="TabsToolbar"
            class="browser-toolbar browser-titlebar"
            fullscreentoolbar="true"
@@ -50,6 +50,8 @@
               tooltip="tabbrowser-tab-tooltip"
               orient="horizontal"
               stopwatchid="tabClick">
+<html:div id="zen-essentials" skipintoolbarset="true"></html:div>
+<html:div id="zen-tabs-wrapper">
           <hbox class="tab-drop-indicator" hidden="true"/>
           <html:span id="tab-drag-empty-feedback" role="presentation"/>
 # If the name (tabbrowser-arrowscrollbox) or structure of this changes
@@ -76,6 +78,7 @@
                               tooltip="dynamic-shortcut-tooltip"
                               data-l10n-id="tabs-toolbar-new-tab"/>
           <html:span id="tabbrowser-tab-a11y-desc" hidden="true"/>
+</html:div>
         </tabs>
 
         <toolbarbutton id="new-tab-button"
@@ -101,9 +104,10 @@
 #include private-browsing-indicator.inc.xhtml
     <toolbarbutton class="content-analysis-indicator toolbarbutton-1 content-analysis-indicator-icon"/>
 
-#include titlebar-items.inc.xhtml
+#include zen-sidebar-icons.inc.xhtml
 
   </toolbar>
+</hbox>
 
   <toolbar id="nav-bar"
            class="browser-toolbar chromeclass-location"
@@ -476,27 +480,31 @@
                      hidden="true"
                      overflows="false"
                      removable="false"/>
-    </hbox>
-
-    <toolbarbutton id="nav-bar-overflow-button"
-                   class="toolbarbutton-1 chromeclass-toolbar-additional"
-                   delegatesanchor="true"
-                   skipintoolbarset="true"
-                   data-l10n-id="navbar-overflow">
-      <box class="toolbarbutton-animatable-box">
-        <image class="toolbarbutton-animatable-image"/>
-      </box>
-    </toolbarbutton>
+    
+
+      <toolbarbutton id="nav-bar-overflow-button"
+                    class="toolbarbutton-1 chromeclass-toolbar-additional"
+                    delegatesanchor="true"
+                    skipintoolbarset="true"
+                    removable="true"
+                    overflows="false"
+                    data-l10n-id="navbar-overflow">
+        <box class="toolbarbutton-animatable-box">
+          <image class="toolbarbutton-animatable-image"/>
+        </box>
+      </toolbarbutton>
 
-    <toolbaritem id="PanelUI-button"
-                 removable="false">
-      <toolbarbutton id="PanelUI-menu-button"
-                     class="toolbarbutton-1"
-                     delegatesanchor="true"
-                     badged="true"
-                     consumeanchor="PanelUI-button"
-                     data-l10n-id="appmenu-menu-button-closed2"/>
-    </toolbaritem>
+      <toolbaritem id="PanelUI-button"
+                  removable="false"
+                  overflows="false">
+        <toolbarbutton id="PanelUI-menu-button"
+                      class="toolbarbutton-1"
+                      delegatesanchor="true"
+                      badged="true"
+                      consumeanchor="PanelUI-button"
+                      data-l10n-id="appmenu-menu-button-closed2"/>
+      </toolbaritem>
+    </hbox>
     <hbox class="titlebar-spacer" type="post-tabs"/>
 #include private-browsing-indicator.inc.xhtml
     <toolbarbutton class="content-analysis-indicator toolbarbutton-1 content-analysis-indicator-icon"/>